src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs

```diff
@@ -569,9 +569,21 @@ private void WriteDispose(INamedTypeSymbol typeSymbol, IndentedStringBuilder bui
 					[SuppressMessage(
 						""Microsoft.Usage"",
 						""CA2215:DisposeMethodsShouldCallBaseClassDispose"",
-						Justification = ""The dispose is re-scheduled using the ValidateDispose method"")]
+						Justification = ""The dispose is re-scheduled in order to properly remove children from their parent"")]
 					protected sealed override void Dispose(bool disposing)
 					{{
+						// This method is present in order to ensure for faster collection of a disposed visual tree
+						// as well as ensure that instances can be properly returned to the FrameworkTemplatePool.
+						//
+						// This method can be called by the finalizer, in which case the object is re-registered 
+						// for finalization, and the Dispose method is invoked explicitly during a idle dispatch.
+						// 
+						// This operation can fail if Dispose() is called by user code on UIView instances.
+						// The Roslyn analyzer UnoDoNotDisposeNativeViews will warn the developer if this is the case.
+						// 
+						// For native exceptions that may be raised if Disposed is called incorrectly, see 
+						// https://github.com/xamarin/xamarin-macios/issues/19493.
+
 						if(_isDisposed)
 						{{
 							base.Dispose(disposing);
```

Comment:
Will this call produce a diagnostic? In this case we might want to skip analyzing generated code.

Reply:
Indeed, and there are other legitimate scenarios where this is useful, so technically, base.Dispose, when called from Dispose it valid.